### ISSUE TITLE
All volumes get faulted randomly

### ISSUE THREAD

**Issue by mtenrero** [longhorn/longhorn#528]
Hello.

I've a simple deployment of k8s with longhorn as base pvc provider on bare-metal. Everything is working fine, excepting sometimes (maybe once or twice per week) all the nodes get faulted automatically without any clear reason.

```
time="2019-05-12T22:38:08Z" level=info msg="Done running ssync [ssync -host 192.168.2.245 -port 9704 volume-snap-7826a6d8-a352-48ac-a5ac-3fe758b9746a.img.meta]"
time="2019-05-12T22:38:08Z" level=info msg="Running ssync [ssync -host 192.168.2.245 -port 9705 volume-snap-c1db11bb-8404-41ad-9966-aa6820a9ad45.img]"
time="2019-05-12T22:38:08Z" level=info msg="Syncing volume-snap-c1db11bb-8404-41ad-9966-aa6820a9ad45.img to 192.168.2.245:9705...\n"
time="2019-05-12T22:38:08Z" level=info msg="source file size: 10737418240, setting up directIo: true"
time="2019-05-12T22:38:30Z" level=info msg="Ssync client: exit code 0"
time="2019-05-12T22:38:30Z" level=info msg="Done running ssync [ssync -host 192.168.2.245 -port 9705 volume-snap-c1db11bb-8404-41ad-9966-aa6820a9ad45.img]"
time="2019-05-12T22:38:30Z" level=info msg="Running ssync [ssync -host 192.168.2.245 -port 9706 volume-snap-c1db11bb-8404-41ad-9966-aa6820a9ad45.img.meta]"
time="2019-05-12T22:38:30Z" level=info msg="Syncing volume-snap-c1db11bb-8404-41ad-9966-aa6820a9ad45.img.meta to 192.168.2.245:9706...\n"
time="2019-05-12T22:38:30Z" level=info msg="source file size: 126, setting up directIo: false"
time="2019-05-12T22:38:30Z" level=warning msg="Failed to open server: 192.168.2.245:9706, Retrying..."
time="2019-05-12T22:38:30Z" level=warning msg="Failed to open server: 192.168.2.245:9706, Retrying..."
time="2019-05-12T22:38:31Z" level=info msg="Ssync client: exit code 0"
```

```
tgtd: bs_longhorn_request(136) unsupported cmd->scb[0]: 35
tgtd: bs_longhorn_request(136) unsupported cmd->scb[0]: 35
tgtd: bs_longhorn_request(136) unsupported cmd->scb[0]: 35
tgtd: bs_longhorn_request(136) unsupported cmd->scb[0]: 35
tgtd: bs_longhorn_request(136) unsupported cmd->scb[0]: 35
tgtd: bs_longhorn_request(136) unsupported cmd->scb[0]: 35
tgtd: bs_longhorn_request(136) unsupported cmd->scb[0]: 35
tgtd: bs_longhorn_request(136) unsupported cmd->scb[0]: 35
tgtd: bs_longhorn_request(136) unsupported cmd->scb[0]: 35
tgtd: bs_longhorn_request(136) unsupported cmd->scb[0]: 35
```

These are some logs from failing pods. Another pods run smoothly:

```
time="2019-05-13T07:07:26Z" level=info msg="Creating volume /volume, size 10737418240/512"
time="2019-05-13T07:07:26Z" level=info msg="Listening on data 0.0.0.0:9503"
time="2019-05-13T07:07:26Z" level=info msg="Listening on sync agent 0.0.0.0:9504"
time="2019-05-13T07:07:26Z" level=info msg="Listening on control 0.0.0.0:9502"
time="2019-05-13T07:07:26Z" level=info msg="Listening on sync 0.0.0.0:9504"
time="2019-05-13T07:07:34Z" level=info msg="New connection from: 192.168.1.55:40644"
time="2019-05-13T07:07:34Z" level=info msg="Opening volume /volume, size 10737418240/512"
192.168.1.55 - - [13/May/2019:07:07:34 +0000] "POST /v1/replicas/1?action=open HTTP/1.1" 200 1254 "" "Go-http-client/1.1"
```

```
192.168.2.179 - - [12/May/2019:22:37:51 +0000] "POST /v1/replicas/dGNwOi8vMTkyLjE2OC4yLjI0NTo5NTAy?action=preparerebuild HTTP/1.1" 200 347 "" "Go-http-client/1.1"
192.168.2.179 - - [12/May/2019:22:38:32 +0000] "GET /v1/replicas/dGNwOi8vMTkyLjE2OC4yLjI0NTo5NTAy HTTP/1.1" 200 432 "" "Go-http-client/1.1"
time="2019-05-12T22:38:32Z" level=info msg="Get backend tcp://192.168.2.244:9502 revision counter 1350852"
time="2019-05-12T22:38:32Z" level=info msg="Set revision counter of 192.168.2.245:9502 to : 1350852"
time="2019-05-12T22:38:32Z" level=info msg="Set backend tcp://192.168.2.245:9502 revision counter to 1350852"
time="2019-05-12T22:38:32Z" level=info msg="WO replica tcp://192.168.2.245:9502's chain verified, update mode to RW"
time="2019-05-12T22:38:32Z" level=info msg="Set replica tcp://192.168.2.245:9502 to mode RW"
192.168.2.179 - - [12/May/2019:22:38:32 +0000] "POST /v1/replicas/dGNwOi8vMTkyLjE2OC4yLjI0NTo5NTAy?action=verifyrebuild HTTP/1.1" 200 432 "" "Go-http-client/1.1"
tgtd: bs_longhorn_request(136) unsupported cmd->scb[0]: 35
```

**Comment by yasker** [comment]
@mtenrero 

The follow log is harmless, we've removed the log from the upcoming release.
```
tgtd: bs_longhorn_request(136) unsupported cmd->scb[0]: 35
```

This log indicates something is wrong with the network
```
time="2019-05-12T22:38:30Z" level=warning msg="Failed to open server: 192.168.2.245:9706, Retrying..."
```

If it doesn't always happen, normally it's due to the CPU load increased on a certain node, cause the engines cannot get enough CPU to process the data, result in volume faulted. If the engine is running on that node, the volume may become faulted; if the replica is running on that node, the engine on another node may not able to communicate with that replica so it may lose a replica and start rebuilding.

You can send me a support bundle so I can help to look into it. We had a bug in v0.4.1 regarding the support bundle, you may need the workaround at https://github.com/rancher/longhorn/issues/414#issuecomment-472131576 .

**Comment by yasker** [comment]
Also, in the next release v0.5.0, we've introduced an experimental feature to allow users to reserve CPU for Longhorn engines, though it has some side effects so we won't make it permanent. We want to use it as a confirmation of the root cause of the issue. You can give it a try when it's released.

The permanent fix is coming in v0.6.0.
